actor/builtin/paymentbroker/paymentbroker.go

```diff
@@ -245,6 +248,10 @@ func (pb *Actor) Redeem(vmctx exec.VMContext, payer address.Address, chid *types
 			return errors.NewFaultError("Expected PaymentChannel from channels lookup")
 		}
 
+		if err := checkCondition(vmctx, channel, condition, redeemerConditionParams); err != nil {
```

Comment:
BLOCKING: This needs to go into `updateChannel` so we run the same validation logic in redeem and close and also so that we can run the validations in the right order. It might be worth renaming `updateChannel` to `validateAndUpdateChannel`.